Task_2.py

```diff
@@ -1,10 +1,105 @@
-import sys
+import json
+phonebook = {}
+def add(phonebook_name, first_name, last_name, number, city_or_country): 
+    with open('my_phonebook.json', 'r') as file:
+        phonebook = json.load(file)
+        if phonebook_name in phonebook: # не працює так як повинно
+            print("Alredy exist")
+        else:
+            phonebook[phonebook_name] = {"First_name": first_name, "Last_name": last_name, "Full_name": first_name + " " + last_name, "number":number, "city_or_country":city_or_country}
+            with open('my_phonebook.json', 'w') as file:
+                json.dump(phonebook, file, indent=4)
+add(11, "a", "b", 4, 5)
+add(25, "a", "b", 4, 6)
 
-sys.path.append("C:\\Users\\User\\OneDrive\\Робочий стіл\\Python\\Beetroot_128\\Home_work_9\\sys.py")
-for i in sys.path:
-    del sys.path[0]
-    del sys.path[0]
-    del sys.path[0]
-print (sys.path)
-# Я можу видалити всі шляхи до sys, окрім пари, з дерикторії в якій я запускаю модулі
-# на функціонал це ніяк не вплинуло
+#add(6, "c", "b", 7, 8)
+#add(9, "g", "w", 11, 12)
+
+def search_by_first_name(first_name):
+    with open('my_phonebook.json', 'r') as file:
+        data = json.load(file)
+        already_in = False
+        for k, v in data.items():
+            if v.get("First_name") == first_name:
+                print (f'Found phonedook {k} and data {v}') 
+                already_in = True
+        if not already_in :
+            print("Not found")
+#search_by_first_name("f")
+def search_by_first_name(first_name):
+    with open('my_phonebook.json', 'r') as file:
+        data = json.load(file)
+        already_in = False
+        for k, v in data.items():
+            if v.get("First_name") == first_name:
+                print (f'Found phonedook {k} and data {v}') 
+                already_in = True
+        if not already_in :
+            print("Not found")
+def search_by_last_name(last_name):
+    with open('my_phonebook.json', 'r') as file:
+        data = json.load(file)
+        already_in = False
+        for k, v in data.items():
+            if v.get("Last_name") == last_name:
+                print (f'Found phonedook {k} and data {v}') 
+                already_in = True
+        if not already_in :
+            print("Not found")
+def search_by_full_name(full_name):
+    with open('my_phonebook.json', 'r') as file:

```

Comment:
try to avoid reading from file on every action

Reply:
done